### PR TITLE
Refactor/exposing feedback

### PR DESCRIPTION
**Pull request by pattishin**
Relating to https://github.com/mmw/react-password-strength/issues/50.

**Resolves:** Feedback from zxcvbn not accessible 
**Description:** Allows react-password-strength to pass back `feedback` object from `zxcvbn` module to `changeCallback`.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,6460 @@
+{
```
Comment (mmw) [inline package-lock.json]: plz remove `package.lock` file, I dont think its necessary for this tool

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,5 @@
 *.log
-/node_modules/
 /dist/
+/node_modules/
```
Comment (mmw) [inline .gitignore]: did you intentionally move this or some IDE tool of yours? no big deal just wondering about the change

File: src/index.js [hunk 3]
```diff
@@ -20,6 +21,7 @@ export default class ReactPasswordStrength extends React.Component {
       score: 0,
       isValid: false,
       password: '',
+      feedback: {}
```
Comment (mmw) [inline src/index.js]: would prefer that we dont keep `feedback` in the state just for the reason of passing it along in the callback on ln 62

File: src/index.js [hunk 4]
```diff
@@ -37,20 +39,24 @@ export default class ReactPasswordStrength extends React.Component {
...
     // always sets a zero score when min length requirement is not met
     // which avoids unnecessary zxcvbn computations (they require quite lots of CPU)
     if (this.isTooShort(password)) {
       score = 0;
+      feedback = {};
     } else {
-      score = zxcvbn(password, userInputs).score;
+      const result = zxcvbn(password, userInputs);
+      score = result.score;
+      feedback = result.feedback;
```
Comment (mmw) [inline src/index.js]: I think what we'd prefer is to calculate the result and save it to a variable that we then pass to the `changeCallback` function. I expect ln 62 to be something like `changeCallback(this.state, result);`

File: src/index.js [hunk 5]
```diff
@@ -37,20 +39,24 @@ export default class ReactPasswordStrength extends React.Component {
     const { changeCallback, minScore, userInputs } = this.props;
     const password = this.refs['ReactPasswordStrength-input'].value;
 
-    let score;
+    let score, feedback;
```
Comment (mmw) [inline src/index.js]: I think the ideal is something like:

```
let score = 0;
let result = null;

if (!this.isTooShort(password) {
  result = zxcvbn(password, userInputs);
  score = results.score;
}

// then in the setState callback (ln 62):

changeCallback(this.state, result);
```

**Review by mmw — CHANGES_REQUESTED** [review]
Thanks for submitting this, please let me know if the feedback/change requests make sense

File: .gitignore (hunk 2 above)
Comment (pattishin) [reply .gitignore]: Ah, this was me adding back in `/dist` after I removed it in my fork to get the small change I originally made to pass back the zxcvbn feedback. I shouldn't have been exposed in this PR. I'm reverting it back to the format you originally had it in.

File: src/index.js (hunk 3 above)
Comment (pattishin) [reply src/index.js]: Makes sense!

**Comment by pattishin** [comment]
Thanks for your comments! Changes have been made. Let me know if you have any other suggestions! 😄

File: src/index.js [hunk 6]
```diff
@@ -37,23 +37,23 @@ export default class ReactPasswordStrength extends React.Component {
...
 
     this.setState({
       isValid: score >= minScore,
       password,
-      score,
+      score
     }, function() {
       if (changeCallback !== null) {
-        changeCallback(this.state);
+        changeCallback(this.state, result);
```
Comment (mmw) [inline src/index.js]: please update the `README` to document that we are passing `changeCallback` the computed `result` and a short blurb about what it is

**Review by mmw — COMMENTED** [review]
Can you update `package.json` version to be `2.2.0`? Just this and the docs change are the last two things. Thank you for your work on this!

File: src/index.js [hunk 7]
```diff
@@ -9,7 +9,7 @@ export default class ReactPasswordStrength extends React.Component {
     this.state = {
       score: 0,
       isValid: false,
-      password: '',
```
Comment (mmw) [inline src/index.js]: oh, also please revert these unnecessary changes (also on ln 22 and 53)

File: src/index.js (hunk 6 above)
Comment (pattishin) [reply src/index.js]: Can do, and reverting those small changes!

**Review by mmw — APPROVED** [review]
